vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -803,19 +803,13 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
 
         grid.$connector.removeFromQueue = tryCatchWrapper(function (item) {
           let itemId = grid.getItemId(item);
+          // The treePageCallbacks for the itemId are about to be discarded ->
+          // Resolve the callbacks with an empty array to not leave grid in loading state
+          Object.values(treePageCallbacks[itemId] || {}).forEach((callback) => callback([]));
```

Comment:
It seems, if we don't provide the size for nested level requests, this [logic](https://github.com/vaadin/web-components/blob/e5c179c5fc67c374b52f24455b026fb242da6ce1/packages/grid/src/vaadin-grid-data-provider-mixin.js#L394-L399) will set the size to the length of the provided items array, which is 0 in this case. I wonder if it can cause any issues. Probably not, considering that `ensureSubCacheForScaledIndex` will anyway request the first page and set the correct size back when the item is expanded again. Just wanted to shine a spotlight on this.

Reply:
I did consider this but concluded it shouldn't be an issue because, as you mentioned, the page will get requested again anyway and the correct size will be provided after the response.